src/iop/colorbalancergb.c

```diff
@@ -202,25 +259,32 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
 
   // Matrices from CIE 1931 2° XYZ D50 to Filmlight grading RGB D65 through CIE 2006 LMS
   const float XYZ_to_gradRGB[3][4] = { { 0.53346004f,  0.15226970f , -0.19946283f, 0.f },
-                                          {-0.67012691f,  1.91752954f,   0.39223917f, 0.f },
-                                          { 0.06557547f, -0.07983082f,   0.75036927f, 0.f } };
+                                       {-0.67012691f,  1.91752954f,   0.39223917f, 0.f },
+                                       { 0.06557547f, -0.07983082f,   0.75036927f, 0.f } };
   const float gradRGB_to_XYZ[3][4] = { { 1.67222161f, -0.11185000f,  0.50297636f, 0.f },
-                                          { 0.60120746f,  0.47018395f, -0.08596569f, 0.f },
-                                          {-0.08217531f,  0.05979694f,  1.27957582f, 0.f } };
+                                       { 0.60120746f,  0.47018395f, -0.08596569f, 0.f },
+                                       {-0.08217531f,  0.05979694f,  1.27957582f, 0.f } };
 
   // Premultiply the pipe RGB -> XYZ and XYZ -> grading RGB matrices to spare 2 matrix products per pixel
   float DT_ALIGNED_ARRAY input_matrix[3][4];
   float DT_ALIGNED_ARRAY output_matrix[3][4];
   mat3mul4((float *)input_matrix, (float *)XYZ_to_gradRGB, (float *)RGB_to_XYZ);
   mat3mul4((float *)output_matrix, (float *)XYZ_to_RGB, (float *)gradRGB_to_XYZ);
 
+  // Test white point of the current space in grading RGB
+  const float white_pipe_RGB[4] = { 1.f, 1.f, 1.f };
+  float white_grading_RGB[4] = { 0.f };
+  dot_product(white_pipe_RGB, input_matrix, white_grading_RGB);
+  const float sum_white = white_grading_RGB[0] + white_grading_RGB[1] + white_grading_RGB[2];
+  for(size_t c = 0; c < 4; c++) white_grading_RGB[c] /= sum_white;
```

Comment:
for_four_channel(c) white_grading_RGB[c] /= sum_white;

Reply:
fixed, although it won't make a difference since that part is executed once per image